internal/core/thirdparty/tantivy/tantivy-binding/src/index.rs

```diff
@@ -0,0 +1,329 @@
+use std::fmt;
+use std::ops::Bound;
+use std::str::FromStr;
+
+use futures::executor::block_on;
+
+use tantivy::collector::TopDocs;
+use tantivy::directory::MmapDirectory;
+use tantivy::query::{Query, RangeQuery, TermQuery, RegexQuery};
+use tantivy::schema::{Field, IndexRecordOption, Schema, TextFieldIndexing, TextOptions, INDEXED};
+use tantivy::{doc, tokenizer, DocAddress, Index, IndexReader, IndexWriter, ReloadPolicy, Term};
+
+use crate::data_type::TantivyDataType;
+use crate::util::make_bounds;
+
+pub struct IndexWrapper {
+    pub field_name: Option<String>,
```

Comment:
all those field should never be none by current implementation.
Remove the Option<*> is a better implementation for IndexWrapper.